opc/resource_storage_object.go

```diff
@@ -0,0 +1,265 @@
+package opc
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-oracle-terraform/storage"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mitchellh/go-homedir"
+)
+
+func resourceOPCStorageObject() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOPCStorageObjectCreate,
+		Read:   resourceOPCStorageObjectRead,
+		Delete: resourceOPCStorageObjectDelete,
+		// TODO: Add Import
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the storage object",
+			},
+			"container": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the storage container",
+			},
+			"content": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "Raw content in string-form of the data",
+				ConflictsWith: []string{"copy_from", "file"},
+			},
+			"file": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "File path for the content to use for data",
+				ConflictsWith: []string{"copy_from", "content"},
+			},
+			"content_disposition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Overrides the behavior of the browser",
+			},
+			"content_encoding": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Set the content-encoding metadata",
+			},
+			"content_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Set the MIME type for the object",
+			},
+			"copy_from": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"content", "file"},
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !strings.Contains(value, "/") {
+						errors = append(errors, fmt.Errorf(
+							"%q does not contain both a container and object name",
+							k))
+					}
+					return
+				},
+			},
+			"delete_at": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "Specify the number of seconds after which the system deletes the object",
```

Comment:
> number of seconds

I don't know the use-case for this, but would it make sense to put this in a larger unit? We also have this in the disk sizes, which are submitted in Bytes but are either MB/GB in the Schema